### PR TITLE
[WIP] [DX] Remove parent node from AbstractRector

### PR DESCRIPTION
**Pull request by TomasVotruba**
Let's see what is missing here to remove the parent node :)

### CODE REVIEW

File: packages/NodeTypeResolver/Node/AttributeKey.php [hunk 1]
```diff
@@ -71,6 +71,8 @@ final class AttributeKey
      * @internal of php-parser, do not change
      * @see https://github.com/nikic/PHP-Parser/pull/681/files
      * @var string
+     *
+     * @api for BC layer now
```
Comment (samsonasik) [inline packages/NodeTypeResolver/Node/AttributeKey.php:75]: comment here is needed that `parent` can still be enabled by use:

```neon
conditionalTags:
    PhpParser\NodeVisitor\ParentConnectingVisitor:
        phpstan.parser.richParserNodeVisitor: true
```

in custom phpstan config for : 

```php
$rectorConfig->phpstanConfig(__DIR__ . '/phpstan.neon.dist');
```

with cautions that will not be reliable on attribute changed after, see removed config at PR:

- https://github.com/rectorphp/rector-src/pull/4415#pullrequestreview-1512291349 

for init parent set.

Comment (TomasVotruba) [reply packages/NodeTypeResolver/Node/AttributeKey.php:75]: Good point, I've added the link to this commen to have easy-to-read context :+1:

**Comment by TomasVotruba** [comment]
@samsonasik I'm looking into this past 2 hours and cannot figure it out. Any ideas why the tests are still failing? 
There should be no parent usage in our code :thinking: 

Seems it's combination of auto import files + removed unused imports option.

**Comment by samsonasik** [comment]
I will check 👍

**Comment by TomasVotruba** [comment]
Just rebased after 2 other successfuly merged PR :+1:

**Comment by TomasVotruba** [comment]
Seems test fixture cache issue with PHPStan, as global file is used. 

I move the content to a class fixture and seems good now :+1:

**Comment by TomasVotruba** [comment]
Surpassed by https://github.com/rectorphp/rector-src/pull/4465

**Pull request closed without merging**